analyzer_test.go

```diff
@@ -347,7 +363,48 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
 
 					files, err := ioutil.ReadDir(layerDir)
 					h.AssertNil(t, err)
-					h.AssertEq(t, len(files), 0)
+					h.AssertEq(t, len(files), 2)
+
+					files, err = ioutil.ReadDir(filepath.Join(layerDir, "metadata.buildpack"))
+					h.AssertNil(t, err)
+					//expect 1 file b/c of store.toml
+					h.AssertEq(t, len(files), 1)
+
+					files, err = ioutil.ReadDir(filepath.Join(layerDir, "no.cache.buildpack"))
+					h.AssertNil(t, err)
+					//expect 1 file b/c of store.toml
+					h.AssertEq(t, len(files), 1)
+				})
+
+				it("restores each store metadata", func() {
+					_, err := analyzer.Analyze(image, testCache)
+					h.AssertNil(t, err)
+					for _, data := range []struct{ name, want string }{
+						// store.toml files.
+						{"metadata.buildpack/store.toml", "[metadata]\n  [metadata.metadata-buildpack-store-data]\n    store-key = \"store-val\""},
+						{"no.cache.buildpack/store.toml", "[metadata]\n  [metadata.no-cache-buildpack-store-data]\n    store-key = \"store-val\""},
+					} {
+						got := h.MustReadFile(t, filepath.Join(layerDir, data.name))
+						h.AssertStringContains(t, string(got), data.want)
+					}
+				})
+
+				when("analyzer is running as root", func() {
+					it.Before(func() {
+						if os.Getuid() != 0 {
+							t.Skip("Skipped when not running as root")
+						}
+					})
```

Comment:
Do tests usually get run as root? I worry that this could sneak past with just a warning. Maybe jsut being paranoid....

Reply:
These do run as root in concourse, which is essentially the only time this particular test gets run. We should probably find a better way to test this behavior in the future.